packages/cli/internal/pkg/aws/cdk/progress.go

```diff
@@ -41,19 +50,145 @@ func (p ProgressStream) DisplayProgress(description string) error {
 	return nil
 }
 
-func runProgressBar(ctx context.Context, description string) chan ProgressEvent {
+func ShowExecution(progressEvents []ProgressStream) []Result {
+	var progressResults []*Result
+	var waitGroup sync.WaitGroup
+	waitGroup.Add(len(progressEvents))
+
+	singleStream, progressResults := combineProgressEvents(progressEvents)
+
+	for event := range singleStream {
+		if event.LastOutput != "" {
+			log.Info().Msg(event.LastOutput)
+		}
+	}
+
+	var results []Result
+	for _, progressResult := range progressResults {
+		results = append(results, *progressResult)
+	}
+
+	return results
+}
+
+func updateResultFromStream(stream ProgressStream, progressResult *Result, wait *sync.WaitGroup) {
+	defer wait.Done()
+	var lastEvent ProgressEvent
+
+	for event := range stream {
+		if event.Err != nil {
+			progressResult.Err = event.Err
```

Comment:
Are we overwriting events data over and over again? (e.g. error will be overwritten and effectively lost.) What is the purpose of this function?